<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="ARPGL000231" name="BasicTests.LocationCheck" tests="4" failures="2" timestamp="19 Jul 2021 06:56:26 GMT" time="0.013" errors="0">
  <testcase name="classes_annoated_with_entity_annotation_should_resides_in_entity_package" time="0.001" classname="BasicTests.LocationCheck">
    <failure type="java.lang.AssertionError" message="Architecture Violation [Priority: MEDIUM] - Rule &#039;classes that are annotated with @Entity should reside in a package &#039;..entity..&#039;&#039; was violated (1 times):
Class &lt;demo.controller.demoController&gt; does not reside in a package &#039;..entity..&#039; in (demoController.java:0)">
      <![CDATA[java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'classes that are annotated with @Entity should reside in a package '..entity..'' was violated (1 times):
Class <demo.controller.demoController> does not reside in a package '..entity..' in (demoController.java:0)
	at com.tngtech.archunit.lang.ArchRule$Assertions.assertNoViolation(ArchRule.java:94)
	at com.tngtech.archunit.lang.ArchRule$Assertions.check(ArchRule.java:82)
	at com.tngtech.archunit.lang.ArchRule$Factory$SimpleArchRule.check(ArchRule.java:198)
	at com.tngtech.archunit.lang.syntax.ObjectsShouldInternal.check(ObjectsShouldInternal.java:81)
	at BasicTests.LocationCheck.classes_annoated_with_entity_annotation_should_resides_in_entity_package(LocationCheck.java:26)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at BasicTests.ExcelGenerate.main(ExcelGenerate.java:25)
]]>
    </failure>
  </testcase> <!-- classes_annoated_with_entity_annotation_should_resides_in_entity_package -->
  <testcase name="interfaces_should_not_placed_in_entity_package" time="0.001" classname="BasicTests.LocationCheck">
    <failure type="java.lang.AssertionError" message="Architecture Violation [Priority: MEDIUM] - Rule &#039;classes that reside in a package &#039;..entity&#039; should not be interfaces&#039; was violated (1 times):
Class &lt;demo.entity.DataRepository&gt; is an interface in (DataRepository.java:0)">
      <![CDATA[java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'classes that reside in a package '..entity' should not be interfaces' was violated (1 times):
Class <demo.entity.DataRepository> is an interface in (DataRepository.java:0)
	at com.tngtech.archunit.lang.ArchRule$Assertions.assertNoViolation(ArchRule.java:94)
	at com.tngtech.archunit.lang.ArchRule$Assertions.check(ArchRule.java:82)
	at com.tngtech.archunit.lang.ArchRule$Factory$SimpleArchRule.check(ArchRule.java:198)
	at com.tngtech.archunit.lang.syntax.ObjectsShouldInternal.check(ObjectsShouldInternal.java:81)
	at BasicTests.LocationCheck.interfaces_should_not_placed_in_entity_package(LocationCheck.java:64)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at BasicTests.ExcelGenerate.main(ExcelGenerate.java:25)
]]>
    </failure>
  </testcase> <!-- interfaces_should_not_placed_in_entity_package -->
  <testcase name="classes_annoated_with_service_annotation_should_resides_in_service_package" time="0.001" classname="BasicTests.LocationCheck"/>
  <testcase name="classes_annoated_with_controller_annotation_should_resides_in_controller_package" time="0.010" classname="BasicTests.LocationCheck"/>
</testsuite> <!-- BasicTests.LocationCheck -->
